src/GoTrueClient.ts

```diff
@@ -251,6 +251,21 @@ export default class GoTrueClient {
     }
   }
 
+  /**
+   * Sets the current user.
+   * @param access_token a jwt access token
+   */
+  setUser(access_token: string): Session {
+    this.currentSession = {
+      ...this.currentSession,
+      access_token,
+      token_type: 'bearer',
+      user: null,
+    }
+
+    return this.currentSession
+  }
```

Comment:
The code in supabase-js works like this:

```js
private _getAuthHeaders(): { [key: string]: string } {
    const headers: { [key: string]: string } = {}
    const authBearer = this.auth.session()?.access_token ?? this.supabaseKey
    headers['apikey'] = this.supabaseKey
    headers['Authorization'] = `Bearer ${authBearer}`
    return headers
  }
```

so really all we need to do is ensure `this.auth.session()?.access_token` will return a JWT.

This function is synchronous, so it isn't doing anything smart like checking the user is valid (perhaps the name of the function isn't great then?)

Reply:
Perhaps this is better as `setAuth()` or `setToken()`?

Reply:
hmm I think any are fine tbh:
- `setUser` suggests how this is most commonly used by most people (but in the future perhaps not all)
- `setAuth` since it's setting the Auth bearer header, I guess we can assume this will always be the case
- `setToken` again makes sense from a consumption point of view, maybe even `setAccessToken`

Reply:
OK will let the guys from the other PR (@ Iamshankhadeep @easy-street @@dbredvick ) give input, then i can change

Reply:
I guess `setAccessToken` or `setAuth` works fine.

Reply:
OK I'll go with `setAuth()` for now - then if we add extra items like refresh_tokens it is more generic